src/main/java/seedu/address/storage/JsonAdaptedPet.java

```diff
@@ -0,0 +1,125 @@
+
+/**
+ * Jackson-friendly version of {@link Pet}.
+ */
+class JsonAdaptedPet {
+
+    public static final String MISSING_FIELD_MESSAGE_FORMAT = "Pet's %s field is missing!";
+
+    private final String name;
+    private final String gender;
+    private final String dateOfBirth;
+    private final String species;
+    private final List<JsonAdaptedFood> foodList = new ArrayList<>();
+    private final List<JsonAdaptedTag> tagged = new ArrayList<>();
+
+    /**
+     * Constructs a {@code JsonAdaptedPet} with the given pet details.
+     */
+    @JsonCreator
+    public JsonAdaptedPet(@JsonProperty("name") String name, @JsonProperty("gender") String gender,
+                          @JsonProperty("dateOfBirth") String dateOfBirth, @JsonProperty("species") String species,
+                          @JsonProperty("foodList") List<JsonAdaptedFood> foodList,
+                          @JsonProperty("tagged") List<JsonAdaptedTag> tagged) {
+        this.name = name;
+        this.gender = gender;
+        this.dateOfBirth = dateOfBirth;
+        this.species = species;
+        if (foodList != null) {
+            this.foodList.addAll(foodList);
+        }
+        if (tagged != null) {
+            this.tagged.addAll(tagged);
+        }
+    }
+
+    /**
+     * Converts a given {@code Pet} into this class for Jackson use.
+     */
+    public JsonAdaptedPet(Pet source) {
+        name = source.getName().fullName;
+        gender = source.getGender().toString();
+        dateOfBirth = source.getDateOfBirth().toString();
+        species = source.getSpecies().toString();
+        foodList.addAll(source.getFoodList().stream()
+                .map(JsonAdaptedFood::new)
+                .collect(Collectors.toList()));
+        tagged.addAll(source.getTags().stream()
+                .map(JsonAdaptedTag::new)
+                .collect(Collectors.toList()));
+    }
+
+    /**
+     * Converts this Jackson-friendly adapted pet object into the model's {@code Pet} object.
+     *
+     * @throws IllegalValueException if there were any data constraints violated in the adapted pet.
+     */
+    public Pet toModelType() throws IllegalValueException {
+        final List<Tag> petTags = new ArrayList<>();
+        for (JsonAdaptedTag tag : tagged) {
+            petTags.add(tag.toModelType());
+        }
+
+        final List<Food> petFoods = new ArrayList<>();
+        for (JsonAdaptedFood food : foodList) {
+            petFoods.add(food.toModelType());
+        }
+
+        if (name == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Name.class.getSimpleName()));
+        }
+        if (!Name.isValidName(name)) {
+            throw new IllegalValueException(Name.MESSAGE_CONSTRAINTS);
+        }
+        final Name modelName = new Name(name);
+
+        if (gender == null) {
+            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Gender.class.getSimpleName()));
```

Comment:
Has Gender.class.getSimpleName() been written? Also did you revert it back to enum? Does it mean we should use another way to get the message instead?

Reply:
iinw [getSimpleName()](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Class.html#getSimpleName()) is a class method of `java.lang.Class` which is built-in

Reply:
According to the example given at  https://www.tutorialspoint.com/java/lang/class_getsimplename.htm
Class Name = seedu.address.model.pet.Gender
Class SimpleName = Gender

Reply:
I changed it to class because I didn't know how to reconstruct the enum from a String value.
That resulted in an error, and from that error I found out how to do so.
Since the enum can be reconstructed I reverted Gender back to an enum.